package/lib/src/beam_guard.dart

```diff
@@ -91,7 +93,7 @@ class BeamGuard {
             return true;
           }
         } else {
-          if (pathBlueprint == location.state.uri.toString()) {
+          if (pathBlueprint == location.state.uri.path) {
             return true;
           }
         }
```

Comment:
here bellow, the line `return regexp.hasMatch(location.state.uri.toString());` should also use `path` instead of `uri.toString()`

Reply:
Passing `uri.toString()` or `uri.path` to `regexp.hasMatch` leads to the same result. I tested it on dartpad:

````dart 
RegExp('/test').hasMatch('/test?query=true'); // --> returns true
````

But yes, for being more consistent I changed it :)